File: files/CH-102.py
import pandas as pd
import numpy as np

path = "CH-102 Compare Rows.xlsx"
input = pd.read_excel(path, usecols = "B:C", skiprows = 1)
test  = pd.read_excel(path, usecols = "G", skiprows = 1, nrows = 12)

result = input.where(input["Sales"] - input["Sales"].shift(1) > 0).dropna().reset_index(drop = True)
result = result["Date"].rename("Dates")
print(result.equals(test["Dates"])) # True

# II Aproach with no comparison using > or < operators

result = input.where(np.sign(input["Sales"] - input["Sales"].shift(1)) == 1).dropna().reset_index(drop = True)
result = result["Date"].rename("Dates")
print(result.equals(test["Dates"])) # True

# be careful and call only first or second approach. :D

File: files/100-199/102/CH-102.R
library(tidyverse)
library(readxl)

path = "files/CH-102 Compare Rows.xlsx"
input = read_excel(path, range = "B2:C26")
test  = read_excel(path, range = "G2:G14")

result = input %>%
  filter(Sales - lag(Sales) > 0) %>%
  select(Dates = Date)

identical(result, test)
# [1] TRUE

# Second approach (no comparison gte or lte)

result = input %>%
  filter(sign(Sales - lag(Sales)) == 1) %>%
  select(Dates = Date)
# [1] TRUE
